gdnative-core/src/core_types/dictionary.rs

```diff
@@ -201,98 +230,86 @@ impl Dictionary<Shared> {
         Dictionary::<Unique>::new().into_shared()
     }
 
-    /// Assume that this is the only reference to this dictionary, on which
-    /// operations that change the container size can be safely performed.
+    #[inline]
+    /// Inserts or updates the value of the element corresponding to the key.
     ///
     /// # Safety
     ///
-    /// It isn't thread-safe to perform operations that change the container
-    /// size from multiple threads at the same time.
-    /// Creating multiple `Unique` references to the same collections, or
-    /// violating the thread-safety guidelines in non-Rust code will cause
-    /// undefined behavior.
-    #[inline]
-    pub unsafe fn assume_unique(self) -> Dictionary<Unique> {
-        let sys = self.sys;
-        std::mem::forget(self);
-
-        Dictionary::<Unique> {
-            sys,
-            _marker: PhantomData,
-        }
+    /// It is only safe to call this on a shared dictionary when no other thread may
+    /// access the underlying collection during the call. Generally, it's not recommended to
+    /// mutate variant collections that may be shared. Prefer using `ThreadLocal` or `Unique`
+    /// collections unless you're absolutely sure that you want this.
+    #[deprecated = "Care should be used when mutating shared variant collections. Prefer ThreadLocal or Unique collections unless you're absolutely sure that you want this."]
```

Comment:
Maybe this can link to `assume_unique` or `assume_thread_local` just as a starting point?

Reply:
Added links.

Reply:
I meant in the deprecation message

Reply:
Added.